src/builder/edit_message.rs

```diff
@@ -254,6 +254,19 @@ impl Builder for EditMessage {
     ) -> Result<Self::Built> {
         self.check_length()?;
 
+        if let Some(user_id) = ctx.2 {
+            #[cfg(feature = "cache")]
```

Comment:
```suggestion
        #[cfg(feature = "cache")]
        if let Some(user_id) = ctx.2 {
```

I think check should not be done at all if the cache feature is not enabled.